### PR TITLE
Feature/20348 leder

### PR DESCRIPTION
**Pull request by cmoesgaard**
Implemented reading and writing of Manager in the usual fashion.
The code for creating new org funk objects has become even more contrived due to having to special handling of 'opgaver' -- it works for now, but will be overhauled in the near future hopefully, along with all of the nearly identical code in the edit functions for each org funk type.

### CODE REVIEW

File: mora/service/common.py [hunk 1]
```diff
@@ -89,21 +89,24 @@ def inactivate_old_interval(old_from: str, old_to: str, new_from: str,
     return payload
 
 
-def set_object_value(obj: dict, path: tuple, vals: List[dict],
+def set_object_value(obj: dict, path: tuple, val: Union[dict, List[dict]],
                      overwrite: bool = False):
     path_list = list(path)
     obj_copy = copy.deepcopy(obj)
 
+    if val is not list:
```
Comment (danchr) [inline mora/service/common.py]: This can't possibly work — don't you mean:

```python
if not isinstance(val, list):
```

…but that suggests you're probably not hitting this case in the tests, so you should either adapt them or drop this clause.

File: mora/service/engagement.py [hunk 2]
```diff
@@ -43,14 +43,15 @@ def list_details(type, id):
...
-    The value above informs you that 'association', 'engagement', 'role' and
-    'leave' are valid for this entry, and that no entry exists at any time
-    for 'association' and 'role', whereas 'engagement' and 'leave' have at
-    least one entry either in the past, present or future.
-
+    The value above informs you that 'association', 'engagement', 'role',
+    'leave' and 'manager' are valid for this entry, and that no entry exists
+    at any time for 'association', 'role' and 'manager', whereas
+    'engagement' and 'leave' have at least one entry either in the past,
+    present or future.
```
Comment (danchr) [inline mora/service/engagement.py]: This is getting rather long-winded, I'd say 🙂 Perhaps we should rephrase it to just describe one or two `true` values, and describe everything else as “the others”…

File: mora/service/engagement.py [hunk 3]
```diff
@@ -266,28 +281,43 @@ def convert_leave(funcid, effect):
...
             keys.FROM: util.to_iso_time(start),
             keys.TO: util.to_iso_time(end),
         }
         return func
 
+    def get_classes(effect):
+        rels = effect['relationer']
+        return [obj.get('uuid') for obj in itertools.chain(
+            rels.get('opgaver', []),
+            rels.get('organisatoriskfunktionstype', [])
```
Comment (danchr) [inline mora/service/engagement.py]: This is in fact fine, but that's not obvious from reading the code. One might think that e.g. `RESPONSIBILITY_FIELD` refers to something other than, say, `MANAGER_LEVEL_FIELD`, but they do not. All these functions are just begging for a `dict` that configures this…

File: mora/service/org.py [hunk 4]
```diff
@@ -142,7 +147,16 @@ def get_organisation(orgid):
...
+    engagements = c.organisationfunktion(tilknyttedeorganisationer=orgid,
+                                         funktionsnavn=keys.ENGAGEMENT_KEY)
+    associations = c.organisationfunktion(tilknyttedeorganisationer=orgid,
+                                          funktionsnavn=keys.ASSOCIATION_KEY)
+    leaves = c.organisationfunktion(tilknyttedeorganisationer=orgid,
+                                    funktionsnavn=keys.LEAVE_KEY)
+    roles = c.organisationfunktion(tilknyttedeorganisationer=orgid,
+                                   funktionsnavn=keys.ROLE_KEY)
+    managers = c.organisationfunktion(tilknyttedeorganisationer=orgid,
+                                      funktionsnavn=keys.MANAGER_KEY)
```
Comment (danchr) [inline mora/service/org.py]: We'll have to keep our eye out for how this affects performance. In particular, do the SQL stored procedures perform the same array-compare when filtering for `funktionsnavn` as they do when filtering for `gyldighed`? If so, we can't afford this granularity, as the cost is prohibitive when counting the 36k engagements in the Aarhus spreadsheet.

Comment (cmoesgaard) [reply mora/service/org.py]: That's a good point. I'll let it stay for now, but keep in mind for next time we run it with a large dataset.
But let's also discuss later whether or not this level of granularity is actually useful information at all

Comment (danchr) [reply mora/service/org.py]: Yeah, the idea is just to present *something* — and let someone else decide whether it's useful at all…
